elastic/kibana
# [Response Ops][Flapping] Rule Specific Flapping - Create/Update API changes

## Summary
Issue: https://github.com/elastic/kibana/issues/190018

Implement rule specific flapping support for create and update Rule API. The new property on the rule is named `flapping`;

```
flapping: {
  look_back_window: number;
  status_change_threshold: number;
}
```

Also make changes in the task runner to use the rule's flapping settings if it exists. Otherwise use the global flapping setting. 

# To test
1. Go to `x-pack/plugins/triggers_actions_ui/public/common/constants/index.ts` and turn `IS_RULE_SPECIFIC_FLAPPING_ENABLED` to `true`
2. Create a rule with a rule specific flapping setting, generate the alert and let it flap
3. Assert that the flapping is now using the rule specific flapping
4. Turn space flapping off
5. Assert that it no longer flaps despite having a rule specific flapping
6. Try deleting/adding back the rule specific flapping via the UI and verify everything works. 

### Checklist
